docs/preview/03-Features/secrets/general.md

```diff
@@ -16,14 +16,41 @@ string secretVersion = secret.Version;
 DateTimeOffset? expirationDate = secret.Expires;
 ```
 
-# Raw secrets
+## Raw secrets
 In some scenarios you'd like to just get the secret value directly without any metadata.
 This is possible by calling the `...Raw...` variants on the `ISecretProvider` implementations.
 
 ```csharp
 string secretValue = await secretProvider.GetRawSecretAsync("EventGrid-AuthKey");
 ```
 
+## Synchronous secrets
+In some scenarios you'd like to retrieve secrets synchronously. A common situation is when you want to register a dependent service into the dependency container that requires a secret, but since such a container only registers instances or synchronous functions, there is no easy way to retrieve a secret asynchronously.
+
+Almost all built-in secret providers we provide support synchronous secret retrieval, only the [HashiCorp Vault secret provider](../secret-store/provider/hashicorp-vault.md) does not support this.
+
+Retrieving synchronous secrets can be done via either using the `ISyncSecretProvider` alternative interface, or by calling the `GetSecret` extensions on an `ISecretProvider` implementation.
+
+```csharp
+var services = new ServiceCollection();
+services.AddSingleton(serviceProvider =>
+{
+    // #1 injecting the `ISyncSecretProvider`:
+    var syncSecretProvider = serviceProvider.GetRequiredService<ISyncSecretProvider>();
+    Secret secret = synSecretProvider.GetSecret("EventGrid-AuthKey");
+    string secretValue = syncSecretProvider.GetRawSecret("EventGrid-AuthKey");
+
+    // #2 calling `GetSecret` or `GetRawSecret` extension on `ISecretProvider`:
```

Comment:
Ok, this extension method makes it easier to use (hence my earlier comment on 'why 2 different interfaces').
I think I now also understand why you decided to separate this functionality in another interface; it's because not all providers support this (like hashicorp).  However, we could've decided to in that case throw a `NotSupportedException`.
(Which is also probably what the extension method will do, if called on a `ISecretProvider` implementation that does not support sync secret retrieval?

Reply:
Yes, `SecretNotFoundException` with an `InvalidOperationException` as inner exception.